app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/recordaudio/AudioMediaRecorder.kt

```diff
@@ -209,13 +222,100 @@ class AudioMediaRecorder @Inject constructor(
+
+        val bufferInfo = MediaCodec.BufferInfo()
+        val muxer = MediaMuxer(mp4OutputPath.toString(), MediaMuxer.OutputFormat.MUXER_OUTPUT_MPEG_4)
+        var trackIndex = -1
+        var sawInputEOS = false
+        var sawOutputEOS = false
+
+        val fileInputStream = FileInputStream(inputFile)
+
+        while (!sawOutputEOS) {
+            if (!sawInputEOS) {
+                val inputBufferIndex = codec.dequeueInputBuffer(TIMEOUT_US)
+                if (inputBufferIndex >= 0) {
+                    val inputBuffer = codec.getInputBuffer(inputBufferIndex)
+                    inputBuffer?.clear()
+
+                    val sampleSize = fileInputStream.channel.read(inputBuffer!!)
+                    if (sampleSize < 0) {
+                        codec.queueInputBuffer(inputBufferIndex, 0, 0, 0, MediaCodec.BUFFER_FLAG_END_OF_STREAM)
+                        sawInputEOS = true
+                    } else {
+                        codec.queueInputBuffer(inputBufferIndex, 0, sampleSize, System.nanoTime() / 1000, 0)
+                    }
+                }
+            }
+
+            val outputBufferIndex = codec.dequeueOutputBuffer(bufferInfo, TIMEOUT_US)
+            if (outputBufferIndex >= 0) {
+                val outputBuffer = codec.getOutputBuffer(outputBufferIndex)
+
+                if (bufferInfo.flags and MediaCodec.BUFFER_FLAG_CODEC_CONFIG != 0) {
+                    codec.releaseOutputBuffer(outputBufferIndex, false)
+                    continue
+                }
+
+                if (bufferInfo.size != 0) {
+                    outputBuffer?.position(bufferInfo.offset)
+                    outputBuffer?.limit(bufferInfo.offset + bufferInfo.size)
+
+                    if (trackIndex == -1) {
+                        val newFormat = codec.outputFormat
+                        trackIndex = muxer.addTrack(newFormat)
+                        muxer.start()
+                    }
+
+                    muxer.writeSampleData(trackIndex, outputBuffer!!, bufferInfo)
+                }
+
+                codec.releaseOutputBuffer(outputBufferIndex, false)
+
+                if (bufferInfo.flags and MediaCodec.BUFFER_FLAG_END_OF_STREAM != 0) {
+                    sawOutputEOS = true
+                }
+            }
+        }
+
+        fileInputStream.close()
+        muxer.stop()
+        muxer.release()
+        codec.stop()
+        codec.release()
+    }
+
     companion object {
         fun getRecordingAudioFileName(): String = "wire-audio-${DateTimeUtil.currentInstant().fileDateTime()}.wav"
+        fun getRecordingMP4AudioFileName(): String = "wire-audio-${DateTimeUtil.currentInstant().fileDateTime()}.mp4"
         fun getRecordingAudioEffectsFileName(): String = "wire-audio-${DateTimeUtil.currentInstant().fileDateTime()}-filter.wav"
 
         const val SIZE_OF_1MB = 1024 * 1024
         const val AUDIO_CHANNELS = 1 // Mono
-        const val SAMPLING_RATE = 44100
+        const val SAMPLING_RATE = 16000
```

Comment:
won't decreasing the sampling rate and decrease the audio quality? 🤔

Reply:
yes, i already talked to design about the UX and we did run some tests, it was not a huge decrease, and it should make the encoding a tiny bit faster